packages/schematics/angular/utility/dependency_spec.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  EmptyTree,
+  Rule,
+  SchematicContext,
+  TaskConfigurationGenerator,
+  Tree,
+  callRule,
+  chain,
+} from '@angular-devkit/schematics';
+import { addDependency } from './dependency';
+
+async function testRule(rule: Rule, tree: Tree): Promise<TaskConfigurationGenerator[]> {
+  const tasks: TaskConfigurationGenerator[] = [];
+  const context = {
+    addTask(task: TaskConfigurationGenerator) {
+      tasks.push(task);
+    },
+  };
+
+  await callRule(rule, tree, context as unknown as SchematicContext).toPromise();
+
+  return tasks;
+}
+
+describe('addDependency', () => {
+  it('adds a package to "dependencies" by default', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        dependencies: {},
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('replaces a package version if already present', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        dependencies: { '@angular/core': '^13.0.0' },
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('adds a package version with other packages in alphabetical order', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify(
+        {
+          dependencies: { '@angular/common': '^14.0.0', '@angular/router': '^14.0.0' },
+        },
+        null,
+        2,
+      ),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readText('/package.json')).toBe(
```

Comment:
Nit: Could we get better error messages from parsing the JSON and comparing the structured data?

```typescript
expect(tree.readJson('/package.json')).toEqual({
  dependencies: jasmine.objectInOrderOrSomething({
    '@angular/common': '^14.0.0',
    '@angular/core': '^14.0.0',
    '@angular/router': '^14.0.0',
  }),
});
```

I'm not sure if that asserts ordering, or if there's a `jasmine.*` utility which would do that. If not, we could do:

```typescript
expect(Object.entries(tree.readJson('/package.json').dependencies)).toEqual([
  [ '@angular/common', '^14.0.0' ],
  [ '@angular/core', '^14.0.0' ],
  [ '@angular/router', '^14.0.0' ],
]);
```